### PR TITLE
Add Dockerfile with required environment for scancode

### PR DESCRIPTION
**Pull request by JeroenKnoops**
# Purpose
Add Dockerfile with scancode requirements.
`Python` is downgraded to 3.6 and `scancode` has been added.

# Actions
* Moved existing `Dockerfile` to `docker` directory
* Add new `Dockerfile.scancode` in `docker` directory

# Related issue.
This has some links with #755

# Test
I've created a new docker image `philipssoftware/tern:2-scancode` for scancode and pushed it to docker hub. ( https://github.com/philips-software/docker-tern )

```
docker run --rm -v /var/run/docker.sock:/var/run/docker.sock philipssoftware/tern:2-scancode report -x scancode -i alpine:latest > alpine.txt
Unable to find image 'philipssoftware/tern:2-scancode' locally
65b3db15f518: Pull complete
850581be87f3: Pull complete
f355d2a41bb6: Pull complete
a72a02a55e5c: Pull complete
cb0d06c13dc6: Pull complete
7bd231b26968: Pull complete
6625882c9040: Pull complete
f57a6d65b5ce: Pull complete
Digest: sha256:2bd5da481fd346959b9cccedcd140e4558a6df523c6f14dba2dae5d5a37f9358
Status: Downloaded newer image for philipssoftware/tern:2-scancode
2020-09-11 14:21:33,159 - DEBUG - __main__ - Starting...
2020-09-11 14:21:33,159 - DEBUG - run - Setting up...
2020-09-11 14:21:33,179 - DEBUG - container - Checking if image "alpine:latest" is available on disk...
2020-09-11 14:21:33,183 - DEBUG - container - Attempting to pull image "alpine:latest"
2020-09-11 14:21:35,827 - DEBUG - container - Image "alpine:latest" downloaded
2020-09-11 14:21:35,827 - DEBUG - container - Checking if image "alpine:latest" is available on disk...
2020-09-11 14:21:35,830 - DEBUG - container - Image "alpine:latest" found
2020-09-11 14:21:36,019 - DEBUG - rootfs - Running command: tar -tf /root/.tern/temp.tar
2020-09-11 14:21:36,023 - DEBUG - rootfs - Running command: tar -x --exclude=.wh.* -f /root/.tern/temp.tar -C /root/.tern/temp
2020-09-11 14:21:36,035 - DEBUG - rootfs - Running command: tar -tf /root/.tern/temp/fa27ccac46e9a5b7ff1d188e99763db1a5fb45adb34a917c1c482dfde99ad432/layer.tar
2020-09-11 14:21:36,040 - DEBUG - rootfs - Running command: tar -x --exclude=.wh.* -f /root/.tern/temp/fa27ccac46e9a5b7ff1d188e99763db1a5fb45adb34a917c1c482dfde99ad432/layer.tar -C /root/.tern/temp/fa27ccac46e9a5b7ff1d188e99763db1a5fb45adb34a917c1c482dfde99ad432/contents
2020-09-11 14:21:36,081 - DEBUG - rootfs - Running command: chmod +x /usr/local/lib/python3.6/site-packages/tern/tools/fs_hash.sh
2020-09-11 14:21:36,089 - DEBUG - rootfs - Running command: /usr/local/lib/python3.6/site-packages/tern/tools/fs_hash.sh /root/.tern/temp/fa27ccac46e9a5b7ff1d188e99763db1a5fb45adb34a917c1c482dfde99ad432/contents
2020-09-11 14:21:36,332 - DEBUG - common - Reading files in filesystem...
2020-09-11 14:21:36,336 - DEBUG - rootfs - Running command: /usr/local/bin/scancode -ilpcu --quiet --timeout 300 --json - /root/.tern/temp/fa27ccac46e9a5b7ff1d188e99763db1a5fb45adb34a917c1c482dfde99ad432/contents
2020-09-11 14:23:39,379 - DEBUG - executor - Collecting file data...
2020-09-11 14:23:39,398 - DEBUG - generator - Creating a detailed report of components in image...
2020-09-11 14:23:39,399 - DEBUG - rootfs - Running command: rm -rf /root/.tern/temp/fa27ccac46e9a5b7ff1d188e99763db1a5fb45adb34a917c1c482dfde99ad432/contents
2020-09-11 14:23:39,410 - DEBUG - run - Teardown...
2020-09-11 14:23:39,417 - DEBUG - rootfs - Running command: rm -rf /root/.tern/temp/mergedir
2020-09-11 14:23:39,420 - DEBUG - rootfs - Running command: rm -rf /root/.tern/temp/workdir
2020-09-11 14:23:39,426 - DEBUG - __main__ - Report completed.
2020-09-11 14:23:39,427 - DEBUG - __main__ - Finished
```
output:
```
$  cat alpine.txt
This report was generated by the Tern Project
Version: 2.2.0

Docker image: alpine:latest:
Generated with ScanCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
OR CONDITIONS OF ANY KIND, either express or implied. No content created from
ScanCode should be considered or used as legal advice. Consult an Attorney
for any legal advice.
ScanCode is a free software code scanning tool from nexB Inc. and others.
Visit https://github.com/nexB/scancode-toolkit/ for support and download.

	Layer 1:
	File licenses found in Layer:  mit AND bsd-new, gpl-2.0, gpl-2.0-plus, bsd-simplified, bsd-new, openssl-ssleay, mpl-2.0
	Packages found in Layer:  None
	Licenses found in Layer:  None
------------------------------------------------

###########################################
# Summary of licenses found in Container: #
###########################################
mit AND bsd-new, gpl-2.0, gpl-2.0-plus, bsd-simplified, bsd-new, openssl-ssleay, mpl-2.0
```

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -6,22 +6,22 @@ FROM debian:buster
...
-    util-linux \
+      attr \
+      findutils \
+      git \
+      gnupg2 \
+      python3 \
+      python3-pip \
+      python3-setuptools \
+      tar \
+      util-linux \
```
Comment (nishakm) [inline docker/Dockerfile:17]: Looks like a tab was accidentally added?

File: docker/Dockerfile.scancode [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2019-2020 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: BSD-2-Clause
+
+FROM python:3.6.12-buster
```
Comment (nishakm) [inline docker/Dockerfile.scancode:4]: Suggest using debian:buster as it's easier to get sources for the packages so we can ship convenience images in the future.

**Review by nishakm — CHANGES_REQUESTED** [review]
@JeroenKnoops Thanks for your PR! I have requested some changes to the code. Plus, could you put the description you have in your PR in the commit message as well? That will make the commit message linter happy :)

File: docker/Dockerfile (hunk 1 above)
Comment (JeroenKnoops) [reply docker/Dockerfile:17]: Normally I would always do some extra indenting after the`apt-get -y install \` line, so it's clear that the lines below belong to the `apt-get` command. This is a matter of taste, I can also remove it.

File: docker/Dockerfile.scancode (hunk 2 above)
Comment (JeroenKnoops) [reply docker/Dockerfile.scancode:4]: Installing python on debian:buster is not trivial. But I agree, using `debian:buster` is better. I will rewrite this.

Comment (JeroenKnoops) [reply docker/Dockerfile.scancode:4]: I'm struggling to get python installed on buster.. sorry for the delay!

Comment (nishakm) [reply docker/Dockerfile.scancode:4]: Why not install python3 using apt?

Comment (JeroenKnoops) [reply docker/Dockerfile.scancode:4]: Scancode requires python3.6 and that does not seem to be available anymore.

**Comment by nishakm** [comment]
@JeroenKnoops  I've managed to install scancode in a python 3.8 environment using `pip install scancode-toolkit[full]`.

**Comment by JeroenKnoops** [comment]
> @JeroenKnoops I've managed to install scancode in a python 3.8 environment using `pip install scancode-toolkit[full]`.

That would be awesome, although the site is still saying scancode is not working with python version above 3.6.x

https://scancode-toolkit.readthedocs.io/en/latest/getting-started/install.html#prerequisites
> ScanCode currently doesn’t support versions above Python 3.6.x, though support will be added soon.

**Comment by nishakm** [comment]
> > @JeroenKnoops I've managed to install scancode in a python 3.8 environment using `pip install scancode-toolkit[full]`.
> 
> That would be awesome, although the site is still saying scancode is not working with python version above 3.6.x
> 
> https://scancode-toolkit.readthedocs.io/en/latest/getting-started/install.html#prerequisites
> 
> > ScanCode currently doesn’t support versions above Python 3.6.x, though support will be added soon.

Looks like soon is here :)
https://github.com/nexB/scancode-toolkit/issues/2312#issuecomment-722687591

**Review by nishakm — APPROVED** [review]

**Comment by nishakm** [comment]
@JeroenKnoops thank you!

**Pull request merged**